Submitted By: Ryan Oliver (ryan dot oliver at pha dot com dot au)
Origin: Ryan Oliver
Date: 2005-04-24
Initial Package Version: 4.0.0
Description: This patch is primarily here to fix issues with libtool when
	building packages on a multi-arch/multilib system, as libtool 
	effectively uses the library search paths from gcc -print-search-dirs
	to locate .la files.
	This poses a problem as by default gcc supplies
		eg: /usr/lib64/gcc/x86_64-pc-linux-gnu/3.4.3/../../../
	which following the example would cause libtool to use .la files under
	/usr/lib64 instead of /usr/lib if building a package 32bit, and thus
	will attempt to link in incompatible 64bit libraries.
	Here we simply remove this path (provided by standard_startfile_prefix)
	from the library search order

--- gcc-4.0.0-orig/gcc/gcc.c	2005-04-16 04:22:57.000000000 +1000
+++ gcc-4.0.0/gcc/gcc.c	2005-04-24 10:07:51.000000000 +1000
@@ -6188,7 +6188,7 @@
 	add_sysrooted_prefix (&startfile_prefixes,
 			      standard_startfile_prefix, "BINUTILS",
 			      PREFIX_PRIORITY_LAST, 0, 1);
-      else if (*cross_compile == '0')
+      /* else if (*cross_compile == '0')
 	{
 	  if (gcc_exec_prefix)
 	    add_prefix (&startfile_prefixes,
@@ -6200,7 +6200,7 @@
 			      machine_suffix,
 			      standard_startfile_prefix, NULL),
 		      NULL, PREFIX_PRIORITY_LAST, 0, 1);
-	}
+	} */
 
       if (*standard_startfile_prefix_1)
  	add_sysrooted_prefix (&startfile_prefixes,
